### PR TITLE
add the new InferenceAnnotatedTypeFactory.

### PR DESCRIPTION
**Pull request by Jianchu**

### CODE REVIEW

File: src/checkers/inference/InferrableAnnotatedTypeFactory.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package checkers.inference;
+
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+
+/**
+ * Interface for indicating that the current the annotatedTypeFactory wish to be
```
Comment (smillst) [inline src/checkers/inference/InferrableAnnotatedTypeFactory.java]: wishes

File: src/checkers/inference/InferrableAnnotatedTypeFactory.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * used by checker-framework-inference. This interface allows a
+ * annotatedTypeFactory to configure its own inference behavior.
```
Comment (smillst) [inline src/checkers/inference/InferrableAnnotatedTypeFactory.java]: an {@link AnnotatedTypeFactory}

File: src/checkers/inference/InferrableAnnotatedTypeFactory.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+public interface InferrableAnnotatedTypeFactory {
+
+    /**
+     * This method could be called if the TreeAnnotator defined in current type
+     * system is needed. Otherwise, InferenceTreeAnnotator will be generated and
+     * used by InferenceAnnotatedTypeFactory.
+     *
+     * @see checkers.inference.InferenceAnnotatedTypeFactory#createTreeAnnotator()
+     *
+     * @return TreeAnnotator, it could be subtype of InferenceTreeAnnotator,
```
Comment (smillst) [inline src/checkers/inference/InferrableAnnotatedTypeFactory.java]: Is InferenceTreeAnnotator a new class that should be apart of this pull request?

File: src/checkers/inference/InferrableAnnotatedTypeFactory.java (hunk 1 above)
Comment (mernst) [inline src/checkers/inference/InferrableAnnotatedTypeFactory.java]: another typo: "the current the"

Comment (mernst) [inline src/checkers/inference/InferrableAnnotatedTypeFactory.java]: And, please don't anthropomorphize an AnnotatedTypeFactory.  (They *hate* it when you do that!)  This indicates a wish of the calling code.

File: src/checkers/inference/InferrableAnnotatedTypeFactory.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * Interface for indicating that the current the annotatedTypeFactory wish to be
+ * used by checker-framework-inference. This interface allows a
+ * annotatedTypeFactory to configure its own inference behavior.
+ */
+
+public interface InferrableAnnotatedTypeFactory {
+
+    /**
+     * This method could be called if the TreeAnnotator defined in current type
```
Comment (smillst) [inline src/checkers/inference/InferrableAnnotatedTypeFactory.java]: This comment describes how this method might be used, but it should describe what this method returns.

Comment (mernst) [inline src/checkers/inference/InferrableAnnotatedTypeFactory.java]: That is, the documentation should first describe what the method does and returns, and then describe the intended use case.

Comment (smillst) [inline src/checkers/inference/InferrableAnnotatedTypeFactory.java]: I'm not sure if the TreeAnnotator returned by this method is allowed to also add \@VarAnnot.

**Comment by wmdietl** [comment]
Created a new PR:
https://github.com/typetools/checker-framework-inference/pull/19

**Pull request closed without merging**